core/src/main/scala/com/nvidia/spark/rapids/tool/analysis/AppSQLPlanAnalyzer.scala

```diff
@@ -56,6 +57,33 @@ class AppSQLPlanAnalyzer(app: AppBase, appIndex: Int) extends AppAnalysisBase(ap
   //      SQLPlanParser.
   var unsupportedSQLPlan: ArrayBuffer[UnsupportedSQLPlan] = ArrayBuffer[UnsupportedSQLPlan]()
   var allSQLMetrics: ArrayBuffer[SQLMetricInfoCase] = ArrayBuffer[SQLMetricInfoCase]()
+  // A map between stage ID and the set of node names
+  val stageToNodeNames: HashMap[Long, Seq[String]] = HashMap.empty[Long, Seq[String]]
+  // A map between stage ID and diagnostic metrics values
+  val stageToDiagnosticMetrics: HashMap[Long, HashMap[String, StatisticsMetrics]] =
+    HashMap.empty[Long, HashMap[String, StatisticsMetrics]]
+
+  /**
+   * Check if the input is one of the diagnostic metric names and update the mapping between
+   * stage ID and diagnostic metrics.
+   * @param stageId    stage ID of the metric
+   * @param metricName name of the metric
+   * @param statistics contains min, median, max and total of the input metric
+   */
+  private def updateStageDiagnosticMetrics(stageId: Long, metricName: String,
+      statistics: StatisticsMetrics): Unit = {
+    metricName match {
+      case MEMORY_SPILLED_METRIC | DISK_SPILLED_METRIC | INPUT_BYTES_READ_METRIC |
```

Comment:
Can we use a helper method to check the presence of `metricName` in a set() instead of individual matching?

Reply:
thanks @parthosa, updated this